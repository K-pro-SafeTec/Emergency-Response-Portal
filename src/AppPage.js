import React from 'react';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';

export default ({ title, Component }) => (
  <React.Fragment>
    <AppBar position="static" color="default">
      <Toolbar>
        <Typography variant="title" color="inherit">{title}</Typography>
      </Toolbar>
    </AppBar>
    <Component />
  </React.Fragment>
);
