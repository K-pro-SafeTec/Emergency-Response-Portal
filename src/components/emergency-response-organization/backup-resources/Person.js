import React from 'react';
import StatusIcon from '../../shared/StatusIcon';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import Typography from '@material-ui/core/Typography';

const Person = ({ status, name }) => (
  <ListItem>
    <ListItemIcon>
      <StatusIcon status={status} />
    </ListItemIcon>
    <ListItemText>
      <Typography variant="body1">{name}</Typography>
    </ListItemText>
  </ListItem>
);

export default Person;
