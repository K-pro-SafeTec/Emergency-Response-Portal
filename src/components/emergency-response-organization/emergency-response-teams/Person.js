import React from 'react';
import ListItem from "@material-ui/core/ListItem/ListItem";
import ListItemText from "@material-ui/core/ListItemText/ListItemText";
import PersonIcon from '@material-ui/icons/Person';

const Person = (person) => {
    return(
      <ListItem>
        <PersonIcon/>
        <ListItemText primary={person.name} />
      </ListItem>
    )
};

export default Person;
