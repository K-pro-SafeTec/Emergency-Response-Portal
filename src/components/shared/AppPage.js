import React from 'react';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import ArrowBack from '@material-ui/icons/ArrowBack';
import Home from '@material-ui/icons/Home';
import SafetecLogo from '../../icons/safetek_for_now.svg';
import { withStyles } from '@material-ui/core';
import { Link } from 'react-router-dom';

const styles = {
  grow: {
    flexGrow: '1',
  },
  homeButton: {
    marginLeft: '-12px',
  },
  backButton: {
    marginRight: '16px',
  },
};

const AppPage = ({ classes, title, back, children }) => (
  <React.Fragment>
    <AppBar position="static" color="default">
      <Toolbar>
        {back && (
          <div>
            <IconButton className={classes.homeButton} component={Link} to="/">
              <Home />
            </IconButton>
            <IconButton className={classes.backButton} component={Link} to={back}>
              <ArrowBack />
            </IconButton>
          </div>
        )}
        <Typography variant="title" color="inherit" className={classes.grow}>
          {title}
        </Typography>
        <a href="https://www.safetec.no/">
          <img
            src={SafetecLogo}
            alt="Levert av Safetec"
            height="40px"
          />
        </a>
      </Toolbar>
    </AppBar>
    {children}
  </React.Fragment>
);

export default withStyles(styles)(AppPage);
