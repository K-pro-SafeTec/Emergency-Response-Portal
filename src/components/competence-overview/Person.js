import React from 'react';
import { Link } from 'react-router-dom';
import { teamById } from '../../dummy-data/team';
import { personById } from '../../dummy-data/person';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import Typography from '@material-ui/core/Typography';
import PersonIcon from '@material-ui/icons/Person';
import StatusIcon from '../shared/StatusIcon';
import AppPage from '../shared/AppPage';
import Status, { mostSevere } from '../../helpers/Status';
import EntityInfo from '../shared/EntityInfo';
import CompetenceStatus from './CompetenceStatus';
import { competenceTypeById } from '../../dummy-data/competenceType';
import Table from '@material-ui/core/Table';
import TableHead from '@material-ui/core/TableHead';
import TableBody from '@material-ui/core/TableBody';
import TableRow from '@material-ui/core/TableRow';
import TableCell from '@material-ui/core/TableCell';

const TeamItem = ({ team, person }) => (
  <ListItem button component={Link} to={`../../teams/${team.id}/`}>
    <ListItemIcon style={{ fontSize: '40px' }}>
      <StatusIcon
        status={
          team.requiredCompetence
            .map(competenceId => (person.competence[competenceId] && person.competence[competenceId].status) || Status.ERROR)
            .reduce(mostSevere) || Status.ERROR
        }
      />
    </ListItemIcon>
    <ListItemText primary={team.name} />
  </ListItem>
);

const Person = ({ match }) => {
  const person = personById[match.params.personId];
  if (person) {
    return (
      <AppPage title="Kompetanseoversikt" back="../..">
        <EntityInfo Icon={PersonIcon}>
          <Typography variant="headline">{person.name}</Typography>
          <Typography variant="subheading">
            {person.roles}
          </Typography>
        </EntityInfo>
        <Table>
          <TableHead>
            <TableRow>
              <TableCell>
                Kompetanse
              </TableCell>
              <TableCell>
                Status
              </TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {Object.keys(person.competence).map(competenceId => (
              <TableRow key={competenceId}>
                <TableCell>
                  {competenceTypeById[competenceId].name}
                </TableCell>
                <TableCell>
                  <CompetenceStatus competence={person.competence[competenceId]} />
                </TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
        <List>
          <ListItem>
            <Typography variant="title">Beredskapslag</Typography>
          </ListItem>
          {person.teams.map(teamId => (
            <TeamItem key={teamId} team={teamById[teamId]} person={person} />
          ))}
        </List>
      </AppPage>
    );
  }
  return <p>Fant ikke noen person med denne IDen.</p>
};

export default Person;
