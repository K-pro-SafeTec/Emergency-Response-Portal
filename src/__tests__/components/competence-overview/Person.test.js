import React from 'react';
import TestRenderer from 'react-test-renderer';
import Person from "../../../components/competence-overview/Person";
import { MemoryRouter as Router } from 'react-router-dom'
import AppPage from "../../../components/shared/AppPage";

test('title is correct', () => {
  const testRenderer = TestRenderer.create(<Router><Person match={{params: {personId: 0}}} /></Router>);
  const testInstance = testRenderer.root;
  expect(testInstance.findByType(AppPage).props.title).toBe("Kompetanseoversikt");
});
