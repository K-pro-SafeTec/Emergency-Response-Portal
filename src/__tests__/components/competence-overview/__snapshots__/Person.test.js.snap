// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <header
    className="MuiPaper-root-14 MuiPaper-elevation4-20 MuiAppBar-root-5 MuiAppBar-positionFixed-6 MuiAppBar-colorDefault-11 mui-fixed"
  >
    <div
      className="MuiToolbar-root-41 MuiToolbar-regular-43 MuiToolbar-gutters-42"
    >
      <a
        className="MuiButtonBase-root-51 MuiIconButton-root-45 AppPage-homeButton-2"
        href="/"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiIconButton-label-50"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-54"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-140"
        />
      </a>
      <button
        className="MuiButtonBase-root-51 MuiButtonBase-disabled-52 MuiIconButton-root-45 MuiIconButton-disabled-49"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="-1"
        type="button"
      >
        <span
          className="MuiIconButton-label-50"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-54"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
        </span>
      </button>
      <button
        className="MuiButtonBase-root-51 MuiButtonBase-disabled-52 MuiIconButton-root-45 MuiIconButton-disabled-49"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="-1"
        type="button"
      >
        <span
          className="MuiIconButton-label-50"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-54"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
            />
          </svg>
        </span>
      </button>
      <h2
        className="MuiTypography-root-63 MuiTypography-title-69 MuiTypography-colorInherit-92 AppPage-grow-1"
      >
        Kompetanseoversikt
      </h2>
      <a
        className="AppPage-safetecLink-3"
        href="https://www.safetec.no/"
      >
        <img
          alt="Levert av Safetec"
          height="44px"
          src="safetec.svg"
          width="153px"
        />
      </a>
    </div>
  </header>,
  <div
    className="AppPage-content-4"
  >
    <div
      className="EntityInfo-container-98"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-54 EntityInfo-avatar-99"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
      <div
        className="EntityInfo-info-100"
      >
        <h1
          className="MuiTypography-root-63 MuiTypography-headline-68"
        >
          Jan Andersen
        </h1>
        <h3
          className="MuiTypography-root-63 MuiTypography-subheading-70"
        >
          Beredskapsleder
        </h3>
      </div>
    </div>
    <table
      className="MuiTable-root-101"
    >
      <thead
        className="MuiTableHead-root-102"
      >
        <tr
          className="MuiTableRow-root-103 MuiTableRow-head-106"
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root-108 MuiTableCell-head-109"
            scope="col"
          >
            Kompetanse
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root-108 MuiTableCell-head-109"
            scope="col"
          >
            Status
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root-116"
      >
        <tr
          className="MuiTableRow-root-103"
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root-108 MuiTableCell-body-110"
          >
            Grunnleggende sikkerhetskurs
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root-108 MuiTableCell-body-110"
          >
            <div>
              <div
                className="CompetenceStatus-icon-117"
              >
                <img
                  alt="Ok"
                  height="35em"
                  src="checkmark.svg"
                  width="35em"
                />
              </div>
              <p
                className="MuiTypography-root-63 MuiTypography-body1-72"
              >
                Gyldig til 02/09/2019
              </p>
            </div>
          </td>
        </tr>
        <tr
          className="MuiTableRow-root-103"
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root-108 MuiTableCell-body-110"
          >
            Trening offshore
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root-108 MuiTableCell-body-110"
          >
            <div>
              <div
                className="CompetenceStatus-icon-117"
              >
                <img
                  alt="Ok"
                  height="35em"
                  src="checkmark.svg"
                  width="35em"
                />
              </div>
              <p
                className="MuiTypography-root-63 MuiTypography-body1-72"
              >
                Gyldig til 10/10/2019
              </p>
            </div>
          </td>
        </tr>
        <tr
          className="MuiTableRow-root-103"
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root-108 MuiTableCell-body-110"
          >
            Øvelse offshore
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root-108 MuiTableCell-body-110"
          >
            <div>
              <div
                className="CompetenceStatus-icon-117"
              >
                <img
                  alt="Advarsel"
                  height="35em"
                  src="yellow_warning.svg"
                  width="35em"
                />
              </div>
              <p
                className="MuiTypography-root-63 MuiTypography-body1-72"
              >
                Syk ved øvelse 23.11.2018
              </p>
            </div>
          </td>
        </tr>
        <tr
          className="MuiTableRow-root-103"
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root-108 MuiTableCell-body-110"
          >
            Beredskapsledelse grunnkurs
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root-108 MuiTableCell-body-110"
          >
            <div>
              <div
                className="CompetenceStatus-icon-117"
              >
                <img
                  alt="Ok"
                  height="35em"
                  src="checkmark.svg"
                  width="35em"
                />
              </div>
              <p
                className="MuiTypography-root-63 MuiTypography-body1-72"
              >
                Gyldig til 07/25/2019
              </p>
            </div>
          </td>
        </tr>
        <tr
          className="MuiTableRow-root-103"
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root-108 MuiTableCell-body-110"
          >
            Samordnet beredskapstrening på land
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root-108 MuiTableCell-body-110"
          >
            <div>
              <div
                className="CompetenceStatus-icon-117"
              >
                <img
                  alt="Ok"
                  height="35em"
                  src="checkmark.svg"
                  width="35em"
                />
              </div>
              <p
                className="MuiTypography-root-63 MuiTypography-body1-72"
              >
                Gyldig til 10/03/2019
              </p>
            </div>
          </td>
        </tr>
        <tr
          className="MuiTableRow-root-103"
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root-108 MuiTableCell-body-110"
          >
            SRC/VHF-sertifikat
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root-108 MuiTableCell-body-110"
          >
            <div>
              <div
                className="CompetenceStatus-icon-117"
              >
                <img
                  alt="Ok"
                  height="35em"
                  src="checkmark.svg"
                  width="35em"
                />
              </div>
              <p
                className="MuiTypography-root-63 MuiTypography-body1-72"
              >
                Gyldig til 07/26/2019
              </p>
            </div>
          </td>
        </tr>
        <tr
          className="MuiTableRow-root-103"
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root-108 MuiTableCell-body-110"
          >
            Ledertrening beredskap og krisehåndtering for plattformsjefer
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root-108 MuiTableCell-body-110"
          >
            <div>
              <div
                className="CompetenceStatus-icon-117"
              >
                <img
                  alt="Ok"
                  height="35em"
                  src="checkmark.svg"
                  width="35em"
                />
              </div>
              <p
                className="MuiTypography-root-63 MuiTypography-body1-72"
              >
                Gyldig til 05/05/2019
              </p>
            </div>
          </td>
        </tr>
        <tr
          className="MuiTableRow-root-103"
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root-108 MuiTableCell-body-110"
          >
            NOFO oljevern-beredskapskurs
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root-108 MuiTableCell-body-110"
          >
            <div>
              <div
                className="CompetenceStatus-icon-117"
              >
                <img
                  alt="Ok"
                  height="35em"
                  src="checkmark.svg"
                  width="35em"
                />
              </div>
              <p
                className="MuiTypography-root-63 MuiTypography-body1-72"
              >
                Gyldig til 07/18/2019
              </p>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
    <ul
      className="MuiList-root-118 MuiList-padding-119"
    >
      <li
        className="MuiListItem-root-122 MuiListItem-default-125 MuiListItem-gutters-129"
        disabled={false}
      >
        <h2
          className="MuiTypography-root-63 MuiTypography-title-69"
        >
          Medlemskompetanse
        </h2>
      </li>
      <a
        className="MuiButtonBase-root-51 MuiListItem-root-122 MuiListItem-default-125 MuiListItem-gutters-129 MuiListItem-button-130"
        href="/teams/1/"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <div
          className="MuiListItemIcon-root-133"
          style={
            Object {
              "fontSize": "40px",
            }
          }
        >
          <img
            alt="Ok"
            height="35em"
            src="checkmark.svg"
            width="35em"
          />
        </div>
        <div
          className="MuiListItemText-root-134"
        >
          <span
            className="MuiTypography-root-63 MuiTypography-subheading-70 MuiListItemText-primary-137"
          >
            Beredskapsledelse
          </span>
        </div>
        <span
          className="MuiTouchRipple-root-140"
        />
      </a>
      <a
        className="MuiButtonBase-root-51 MuiListItem-root-122 MuiListItem-default-125 MuiListItem-gutters-129 MuiListItem-button-130"
        href="/teams/7/"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <div
          className="MuiListItemIcon-root-133"
          style={
            Object {
              "fontSize": "40px",
            }
          }
        >
          <img
            alt="Ok"
            height="35em"
            src="checkmark.svg"
            width="35em"
          />
        </div>
        <div
          className="MuiListItemText-root-134"
        >
          <span
            className="MuiTypography-root-63 MuiTypography-subheading-70 MuiListItemText-primary-137"
          >
            Beredskapsleder
          </span>
        </div>
        <span
          className="MuiTouchRipple-root-140"
        />
      </a>
    </ul>
  </div>,
]
`;
